app/lib/common/widgets/room/select_room_drawer.dart

```diff
@@ -0,0 +1,187 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+
+import 'package:acter/common/widgets/room/brief_room_list_entry.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:path/path.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+// ChildRoomType configures the sub child type of the `Spaces`
+enum RoomType {
+  groupChat,
+  space,
+}
+
+class SelectRoomDrawer extends ConsumerStatefulWidget {
+  final String canCheck;
+  final String? currentSpaceId;
+  final Widget title;
+  final String keyPrefix;
+  final DisplayMode avatarDisplayMode;
```

Comment:
I feel like this param can be determined using ```roomType``` internally and doesn't necessarily need to be passed down...